### PR TITLE
Enable Service endpoints and Network Policy Enforcement to support Private Endpoints

### PR DESCRIPTION
**Pull request by chrisorisawayi**
### Change description ###

Adding support for service endpoints and enforcing endpoint network policies. This is required to support the use of private endpoints by MI Data Platform.

The Terraform azurerm_subnet resource provider requires 'enforce_private_link_endpoint_network_policies' set to 'true' to deploy a private endpoint on a given subnet.

Network module already updated in [this PR to](https://github.com/hmcts/aks-module-network/pull/5) support the change.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: terraform/aks/01-network/20-network.tf [hunk 1]
```diff
@@ -12,10 +12,13 @@ module "network" {
...
   service_shortname     = var.service_shortname
 
-  aks_00_subnet_cidr_blocks              = var.aks_00_subnet_cidr_blocks #UK South
-  aks_01_subnet_cidr_blocks              = var.aks_01_subnet_cidr_blocks #UK West # Currently both clusters in UK South
-  application_gateway_subnet_cidr_blocks = var.application_gateway_subnet_cidr_blocks
-  iaas_subnet_cidr_blocks                = var.iaas_subnet_cidr_blocks
+  aks_00_subnet_cidr_blocks                                   = var.aks_00_subnet_cidr_blocks #UK South
+  aks_01_subnet_cidr_blocks                                   = var.aks_01_subnet_cidr_blocks #UK West # Currently both clusters in UK South
+  application_gateway_subnet_cidr_blocks                      = var.application_gateway_subnet_cidr_blocks
+  iaas_subnet_cidr_blocks                                     = var.iaas_subnet_cidr_blocks
```
Comment (timja) [inline terraform/aks/01-network/20-network.tf:18]: add all the endpoints here ? so we don't need to add them everywhere? you could probably just add them to the module to be honest...

**Review by timja — CHANGES_REQUESTED** [review]
this will only work in sandbox because you've made the new vars required inputs, we are trying to avoid environment specific configuration. if it's just for your PR fine but it shouldn't be merged as it

File: terraform/aks/01-network/20-network.tf (hunk 1 above)
Comment (chrisorisawayi) [reply terraform/aks/01-network/20-network.tf:18]: Wasn't too sure about adding values in the module but agreed best to add them in there so will do.

Comment (chrisorisawayi) [reply terraform/aks/01-network/20-network.tf:18]: Now added in the module.

**Review by timja — CHANGES_REQUESTED** [review]
this PR shouldn't be needed any more right?

**Comment by chrisorisawayi** [comment]
PR no longer needed following module update.

**Pull request closed without merging**